src/cart/cart.controller.ts

```diff
@@ -0,0 +1,74 @@
+import express, { Response, Request } from "express";
+import { CartService } from "./cart.service";
+import { getUserIdFromHeaders } from "../utils";
+import { checkUpdateCartReq } from "../middlewares/updateCart.valodator";
+
+const router = express.Router();
+
+router.get('/', async (req: Request, res: Response) => {    
+    try {
+        const userId = getUserIdFromHeaders(req);      
```

Comment:
I guess you already handle this logic in `authUser` middleware, thus this logic looks redundant in this place